cdci_data_analysis/analysis/instrument.py

```diff
@@ -339,6 +349,7 @@ def get_par_by_name(self,par_name):
 
         for _query in self._queries_list:
             if par_name in _query.par_names:
+                # TODO: this picks the last one if there are many?..
                 p  =  _query.get_par_by_name(par_name)
```

Comment:
Yes. But can we have actually two parameters with the same fir a given query?

Reply:
Nothing prevents that. Also, many `query` objects are scanned here. I made a test [here](https://github.com/oda-hub/dispatcher-app/blob/d2808b2d9bd1948b48be28963a23cb0732f3473f/tests/test_unit.py#L48-L55) for curiosity.
No point to fix it here. It could be addressed and fixed separately. It's not important, it's just an unfortunate design.
Parameter name does not need to be globally unique.